src/sentry/search/events/builder/metrics.py

```diff
@@ -159,7 +164,14 @@ def _get_on_demand_metric_spec(self, field: str) -> OnDemandMetricSpec | None:
 
         groupby_columns = self._get_group_bys()
 
-        if not should_use_on_demand_metrics(self.dataset, field, self.query, groupby_columns):
+        if not should_use_on_demand_metrics(
+            dataset=self.dataset,
+            aggregate=field,
+            query=self.query,
+            groupbys=groupby_columns,
+            query_builder_call=True,
+            use_next_spec_version=self.use_next_spec_version,
```

Comment:
This is the only place where we call `should_use_on_demand_metrics` with these two parameters, thus, controlling when a function is considered or not (vs when we just want to extract).